meters.go

```diff
@@ -0,0 +1,79 @@
+package metrics
+
+import (
+	"math"
+	"time"
+)
+
+// Meters stores the built-in function meters used by default for metrics collection.
+// You can define your custom meter functions via metrics.AddMeter() or metrics.SetMeters().
+var Meters = []MeterFunc{
+	MeterNumberOfRequests,
+	MeterResponseStatus,
+	MeterRequestOperation,
+	MeterResponseTime,
+	MeterResponseBodySize,
+	MeterRequestBodySize,
+}
+
+// MeterNumberOfRequests is used to register the total number of served requests.
+func MeterNumberOfRequests(i *Info, m *Metrics) {
+	m.Counter("req.total").Add()
+}
+
+// MeterResponseStatus is used to count the response status code by range (2xx, 4xx, 5xx).
+func MeterResponseStatus(i *Info, m *Metrics) {
+	s := i.Status / 100
+	if s >= 2 && s < 4 {
+		m.Counter("res.status.ok").Add()
+	}
+	if s == 5 {
+		m.Counter("res.status.error").Add()
+	}
+	if s == 4 {
+		m.Counter("res.status.bad").Add()
+	}
+}
+
+// MeterRequestOperation is used to count the number of request by HTTP operation.
+// Operation in inferred by HTTP verb:
+//
+// - GET, HEAD = read operation
+// - POST, PUT, PATCH, DELETE = write operation
+func MeterRequestOperation(i *Info, m *Metrics) {
+	if i.Request.Method == "GET" || i.Request.Method == "HEAD" {
+		m.Counter("req.reads").Add()
+	}
```

Comment:
It's not about about performance but more about the fact that when I read `else if` I immediately understand that the two case are mutually exclusive.